### PR TITLE
chore: preallocate buffer for rlp

### PR DESCRIPTION
**Pull request by mattsse**

### CODE REVIEW

File: crates/primitives/src/transaction/mod.rs [hunk 1]
```diff
@@ -819,7 +819,8 @@ impl TransactionSignedNoHash {
     /// Calculates the transaction hash. If used more than once, it's better to convert it to
     /// [`TransactionSigned`] first.
     pub fn hash(&self) -> B256 {
-        let mut buf = Vec::new();
+        // pre-allocate buffer for the transaction
+        let mut buf = Vec::with_capacity(128 + self.transaction.input().len());
```
Comment (rkrasiuk) [inline crates/primitives/src/transaction/mod.rs:823]: can we add `#[inline] fn max_transaction_rlp_len` to primitives and document it properly?

Comment (mattsse) [reply crates/primitives/src/transaction/mod.rs:823]: there's no such function

do you mean add it?
could be misleading, i don't even know where the 128 are coming from

Comment (rkrasiuk) [reply crates/primitives/src/transaction/mod.rs:823]: i meant we should add it. no clarity behind this pre-allocation is the main motivation for the request. @joshieDo you prob know where the `128` is coming from? regardless, not a blocker for this PR

**Review by rkrasiuk — APPROVED** [review]

File: crates/primitives/src/transaction/mod.rs (hunk 1 above)
Comment (onbjerg) [reply crates/primitives/src/transaction/mod.rs:823]: agree wrt magic 128. if anything should prob be a constant with docs

**Review by onbjerg — APPROVED** [review]

**Pull request merged**